stop caring about generated code and variants

webern opened the pull request:
<!--
Tips:
- Please read CONTRIBUTING.md to understand our process and our requests for PRs.
- Please file an issue before creating a PR so we can discuss the change and confirm it's not already being worked on.
-->

**Issue number:**

Closes #193

- [x] Requires https://github.com/bottlerocket-os/bottlerocket/pull/4038 (DRAFT until then)

**Description of changes:**

stop caring about the models and logdog dirs

With https://github.com/bottlerocket-os/bottlerocket/pull/4038 we can
finally stop mounting the models directory (and a vestige of logdog
could also be removed).

**Testing done:**

- [x] Checked out  https://github.com/bottlerocket-os/bottlerocket/pull/4038 and built it
- [x] Local tests including build-kit tests pass

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

webern committed:
stop caring about the models and logdog dirs

With https://github.com/bottlerocket-os/bottlerocket/pull/4038 we can
finally stop mounting the models directory (and a vestige of logdog
could also be removed).

bcressey commented on twoliter/embedded/build.Dockerfile:
```diff
@@ -150,12 +150,10 @@ RUN --mount=target=/host \
 
-# We use the "nocache" writable space to generate code where necessary, like the variant-
-# specific models.
+# We use the "nocache" writable space to generate code where necessary.
```
You can also get rid of the `variantcache` stage above

webern replied:
Done. Is the comment outdated as well? Just delete that comment?

bcressey replied:
> Is the comment outdated as well? Just delete that comment?

You can delete it. I'm not sure what "nocache" refers to, probably a slightly older way we generated that mount.

webern committed:
buildsys: remove package variant sensitivity

Remove the logic by which packages are differentially built based on
which variant they are for.
Requires https://github.com/bottlerocket-os/bottlerocket/pull/4038

webern commented:
:arrow_up: Also removed the Variant manifest read during package builds. Re-tested.

bcressey commented on tools/buildsys/src/builder.rs:
```diff
@@ -362,7 +348,6 @@ impl DockerBuild {
                 OutputCleanup::BeforeBuild,
             ),
             target_build_args: TargetBuildArgs::Package(PackageBuildArgs {
-                image_features,
```
We can eliminate all of the `variant*` args too right?

webern replied:
Woah, that's crazy talk. Let me try.

webern commented:
⬆️ Also removed the variant* args. Re-tested. Unfortunately, this push https://github.com/bottlerocket-os/twoliter/compare/8a8f8c6c5536fc316e2a4a3f034802c99d3afcee..34d406fcbcc3973d9615fde99ba632688c6e09ac  caused this error.

```
  #16 [rpmbuild 6/6] RUN --mount=source=.cargo,target=/home/builder/.cargo     --mount=type=cache,target=/home/builder/.cache,from=cache,source=/cache     --mount=source=sources,target=/home/builder/rpmbuild/BUILD/sources     rpmbuild -bb --clean       --undefine _auto_set_build_flags       --define "_target_cpu ${ARCH}"       --define "dist ${BUILD_ID_TIMESTAMP}.${BUILD_ID//-dirty/}.br1"       rpmbuild/SPECS/${PACKAGE}.spec
```


```
  #16 6.555 RPM build errors:
  #16 6.558 error: /etc/rpm/macros: line 368: Macro %_cross_variant has empty body
  #16 6.558 error: /etc/rpm/macros: line 369: Macro %_cross_variant_platform has empty body
  #16 6.558 error: /etc/rpm/macros: line 370: Macro %_cross_variant_runtime has empty body
  #16 6.558 error: /etc/rpm/macros: line 371: Macro %_cross_variant_family has empty body
  #16 6.559 error: /home/builder/.rpmmacros: line 368: Macro %_cross_variant has empty body
  #16 6.559 error: /home/builder/.rpmmacros: line 369: Macro %_cross_variant_platform has empty body
  #16 6.559 error: /home/builder/.rpmmacros: line 370: Macro %_cross_variant_runtime has empty body
  #16 6.559 error: /home/builder/.rpmmacros: line 371: Macro %_cross_variant_family has empty body
  #16 ERROR: process "/bin/sh -c rpmbuild -bb --clean       --undefine _auto_set_build_flags       --define \"_target_cpu ${ARCH}\"       --define \"dist ${BUILD_ID_TIMESTAMP}.${BUILD_ID//-dirty/}.br1\"       rpmbuild/SPECS/${PACKAGE}.spec" did not complete successfully: exit code: 1
  ------
   > [rpmbuild 6/6] RUN --mount=source=.cargo,target=/home/builder/.cargo     --mount=type=cache,target=/home/builder/.cache,from=cache,source=/cache     --mount=source=sources,target=/home/builder/rpmbuild/BUILD/sources     rpmbuild -bb --clean       --undefine _auto_set_build_flags       --define "_target_cpu ${ARCH}"       --define "dist ${BUILD_ID_TIMESTAMP}.${BUILD_ID//-dirty/}.br1"       rpmbuild/SPECS/${PACKAGE}.spec:
  6.555
  6.555 RPM build errors:
  6.558 error: /etc/rpm/macros: line 368: Macro %_cross_variant has empty body
  6.558 error: /etc/rpm/macros: line 369: Macro %_cross_variant_platform has empty body
  6.558 error: /etc/rpm/macros: line 370: Macro %_cross_variant_runtime has empty body
  6.558 error: /etc/rpm/macros: line 371: Macro %_cross_variant_family has empty body
  6.559 error: /home/builder/.rpmmacros: line 368: Macro %_cross_variant has empty body
  6.559 error: /home/builder/.rpmmacros: line 369: Macro %_cross_variant_platform has empty body
  6.559 error: /home/builder/.rpmmacros: line 370: Macro %_cross_variant_runtime has empty body
  6.559 error: /home/builder/.rpmmacros: line 371: Macro %_cross_variant_family has empty body
  ------
  build.Dockerfile:147
  --------------------
   146 |     USER builder
   147 | >>> RUN --mount=source=.cargo,target=/home/builder/.cargo \
   148 | >>>     --mount=type=cache,target=/home/builder/.cache,from=cache,source=/cache \
   149 | >>>     --mount=source=sources,target=/home/builder/rpmbuild/BUILD/sources \
   150 | >>>     # The dist tag is set as the `Release` field in Bottlerocket RPMs. Define it to be
   151 | >>>     # in the form <timestamp of latest commit>.<latest commit short sha>.br1
   152 | >>>     # Remove '-dirty' from the commit sha: '-' is an illegal character for the Release field
   153 | >>>     # and '-dirty' may not be accurate to the state of the actual package being built.
   154 | >>>     rpmbuild -bb --clean \
   155 | >>>       --undefine _auto_set_build_flags \
   156 | >>>       --define "_target_cpu ${ARCH}" \
   157 | >>>       --define "dist ${BUILD_ID_TIMESTAMP}.${BUILD_ID//-dirty/}.br1" \
   158 | >>>       rpmbuild/SPECS/${PACKAGE}.spec
   159 |
  --------------------
```

webern commented:
⬆️ Attempt to fix based on Ben's diff https://github.com/bottlerocket-os/twoliter/compare/34d406fcbcc3973d9615fde99ba632688c6e09ac..2b89efd2ab862e5c3ece5a9951b186392038f104

webern commented:
⬆️ Attempt to fix the fix https://github.com/bottlerocket-os/twoliter/compare/2b89efd2ab862e5c3ece5a9951b186392038f104..bcc1a60e4f4ddf2cf8f951b621fcd69ecd7f5d45

webern commented:
⬆️ Attempt to fix the fix of the fix https://github.com/bottlerocket-os/twoliter/compare/bcc1a60e4f4ddf2cf8f951b621fcd69ecd7f5d45..04723616fc054e4626e7800bf757af525e315eba

bcressey commented on twoliter/embedded/build.Dockerfile:
```diff
@@ -114,18 +52,16 @@ USER builder
+   cp "/usr/lib/rpm/platform/${ARCH}-bottlerocket/macros" .rpmmacros \
    && echo "Epoch: ${BUILD_EPOCH}" >> rpmbuild/SPECS/${PACKAGE}.spec \
-   && cat generated.bconds ${PACKAGE}.spec >> rpmbuild/SPECS/${PACKAGE}.spec \
+   && cp ${PACKAGE}.spec rpmbuild/SPECS/${PACKAGE}.spec \
```
need to reorder this above the previous line, or we'll miss out on the `Epoch` for the top-level package

webern replied:
Oh, I see I was blowing it away... this should fix it right? https://github.com/bottlerocket-os/twoliter/compare/224bc50b442bc4f82c3111f955f215d7bc79b195..2721a636b44a41aefdf59a6c9c6c252b1244eb02

webern commented:
⬆️ Attempt this push finally got it to work! Bottlerocket build succeeded https://github.com/bottlerocket-os/twoliter/compare/04723616fc054e4626e7800bf757af525e315eba..224bc50b442bc4f82c3111f955f215d7bc79b195

webern commented:
⬆️ Fix the Epoch bug found by Ben https://github.com/bottlerocket-os/twoliter/compare/224bc50b442bc4f82c3111f955f215d7bc79b195..2721a636b44a41aefdf59a6c9c6c252b1244eb02

webern commented:
Bummer

```
  34.21     File must begin with "/": Epoch:
  34.21     File must begin with "/": 1
```

webern commented:
⬆️ Make `Epoch: 1` the first thing in the RPM spec file. https://github.com/bottlerocket-os/twoliter/compare/2721a636b44a41aefdf59a6c9c6c252b1244eb02..4ff4ec58bd71d768f431729e248ffe8a5f2fce96

webern commented:
⬆️ rebase https://github.com/bottlerocket-os/twoliter/compare/4ff4ec58bd71d768f431729e248ffe8a5f2fce96..33185dc970442042d77735bf0b497769d4048057

webern commented:
Uh oh, after rebasing I got this. Not sure if my conflict resolution was wrong or what...

```
Unable to instantiate the builder: Failed to create build arguments due to a dependency error: Failed to read external kit metadata file '/home/brigmatt/repos/bottlerocket/build/external-kits/external-kit-metadata.json': No such file or directory (os error 2)
error: failed to run custom build command for `filesystem v0.1.0 (/home/brigmatt/repos/bottlerocket/packages/filesystem)`
```

webern commented:
Oh, I think I know. The Bottlerocket Makefile.toml is going to have to call twoliter update and twoliter fetch

webern commented:
Hmmm in Bottlerocket...

```
$ ./tools/twoliter/twoliter fetch
[2024-06-13T00:08:39Z WARN  twoliter::project] A Release.toml file was found. Release.toml is deprecated. Please remove it from your project.
Error: failed to inspect manifest of resource at public.ecr.aws/bottlerocket/bottlerocket-sdk:v0.41.0
```

jmt-lab reviewed (approved):
LGTM
